DOC fixes for LogisticRegression newton-cholesky and multiclass (#31410)
scikit-learn/scikit-learn: doc/modules/linear_model.rst

+------------------------------+-------------+-----------------+-----------------+-----------------------+-----------+------------+
| **Multiclass support**       |                                                                                                  |
+------------------------------+-------------+-----------------+-----------------+-----------------------+-----------+------------+

> | multinomial multiclass       |     yes     |       no        |       yes       |     no                |    yes    |    yes     |

Reviewer: Hopefully https://github.com/scikit-learn/scikit-learn/pull/31410/files#r2102035305 will fix this.

> | multinomial multiclass       |     yes     |       no        |       yes       |     no                |    yes    |    yes     |

Author: Thanks for the fix.

| Robust to unscaled datasets  |     yes     |       yes       |       yes       |     yes               |    no     |    no      |
+------------------------------+-------------+-----------------+-----------------+-----------------------+-----------+------------+

The "lbfgs" solver is used by default for its robustness. For
`n_samples >> n_features`, "newton-cholesky" is a good choice and can reach high

> precision (tiny `tol` values). For large datasets
> the "saga" solver is usually faster.

Author: ```suggestion
precision (tiny `tol` values). For large datasets
the "saga" solver is usually faster (than "lbfgs"), in particular for low precision
(high `tol`).
```

> The "lbfgs" solver is used by default for its robustness. For large datasets
> the "saga" solver is usually faster.
> For `n_samples >> n_features`, "newton-cholesky" is a good choice. In addition, it can
> reach high precision (tiny `tol` values).

Author: ```suggestion
The "lbfgs" solver is used by default for its robustness. For
`n_samples >> n_features`, "newton-cholesky" is a good choice and can reach high
precision (tiny `tol` values). For large datasets
the "saga" solver is usually faster.
```
I think this reads a bit better because the sentence about SGD then comes after saga as it was meant to be originally, I guess.
